### PR TITLE
Setup swap

### PR DESCRIPTION
**Pull request by thomaseizinger**

### CODE REVIEW

**Review by rishflab — APPROVED** [review]

File: comit/src/network/protocols/setup_swap.rs [hunk 1]
```diff
@@ -0,0 +1,646 @@
...
+                    network: Network::Bitcoin,
+                    asset: common.bitcoin_quantity,
+                    redeem_identity: bob.bitcoin_identity,
+                    refund_identity: alice.bitcoin_identity,
+                    expiry: Timestamp::from(common.bitcoin_absolute_expiry),
+                    secret_hash: alice.secret_hash,
+                },
+                beta: herc20::Params {
+                    asset: asset::Erc20 {
+                        token_contract: Default::default(),
```
Comment (thomaseizinger) [inline comit/src/network/protocols/setup_swap.rs:68]: Token contract should be part of `CommonParams` I believe :)

File: comit/src/network/protocols/setup_swap.rs [hunk 2]
```diff
@@ -0,0 +1,646 @@
...
+                        quantity: common.ethereum_quantity,
+                    },
+                    redeem_identity: alice.ethereum_identity,
+                    refund_identity: bob.ethereum_identity,
+                    expiry: Timestamp::from(common.ethereum_absolute_expiry),
+                    secret_hash: alice.secret_hash,
+                    chain_id: ChainId::regtest(),
```
Comment (thomaseizinger) [inline comit/src/network/protocols/setup_swap.rs:75]: I _think_ the chain ID and bitcoin network should be part of the `CommonParams` as well. We can retrieve them from the config when we convert the match and trigger `SetupSwap`. I feel it is worthwhile "asserting" them as well with the other party. What do you think?

File: comit/src/network/protocols/setup_swap.rs [hunk 3]
```diff
@@ -0,0 +1,646 @@
...
+        _: &Self::Protocol,
+        _io: &mut T,
+    ) -> io::Result<Self::Response>
+    where
+        T: AsyncRead + Unpin + Send,
+    {
+        Ok(())
+    }
+
+    /// Writes a take order request to the given I/O stream.
```
Comment (thomaseizinger) [inline comit/src/network/protocols/setup_swap.rs:539]: This comment is a bit outdated :grimacing:

File: comit/src/network/protocols/setup_swap.rs [hunk 4]
```diff
@@ -0,0 +1,646 @@
...
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct CommonParams {
+    pub ethereum_quantity: asset::Erc20Quantity,
+    #[serde(with = "asset::bitcoin::sats_as_string")]
+    pub bitcoin_quantity: asset::Bitcoin,
+    pub ethereum_absolute_expiry: u32,
+    pub bitcoin_absolute_expiry: u32,
+}
+
+impl Display for CommonParams {
```
Comment (thomaseizinger) [inline comit/src/network/protocols/setup_swap.rs:466]: What required `Display` to be implemented? Forwarding to `Debug` is not that useful :)

File: comit/src/network/protocols/setup_swap.rs [hunk 5]
```diff
@@ -0,0 +1,646 @@
...
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::{HashMap, VecDeque},
+    fmt::{Debug, Display},
+    io,
+    marker::PhantomData,
+    task::{Context, Poll},
+};
+
+/// Event emitted by the `Comit` behaviour.
```
Comment (thomaseizinger) [inline comit/src/network/protocols/setup_swap.rs:22]: Stale comment :)

**Review by thomaseizinger — COMMENTED** [review]
Well done, thanks for sticking with the changes we discussed :)

Made a few more minor comments!

I can't approve my own PR though :joy: 
Maybe close this one an open a new one? :)

**Comment by thomaseizinger** [comment]
+ You might want to squash this into one commit :)

File: comit/src/network/protocols/setup_swap.rs (hunk 2 above)
Comment (rishflab) [reply comit/src/network/protocols/setup_swap.rs:75]: yep im cool with that, should be at some point and I guess it is related to the swap.

**Review by rishflab — APPROVED** [review]

**Comment by rishflab** [comment]
Reopening under my name because @thomaseizinger cannot review

**Pull request closed without merging**